src/index.ts

```diff
@@ -366,22 +360,31 @@ export class MayanRoute<N extends Network>
   }
 
   public override async *track(receipt: R, timeout?: number) {
+    if (isCompleted(receipt) || isRedeemed(receipt)) return receipt;
+
     // What should be the default if no timeout is provided?
     let leftover = timeout ? timeout : 60 * 60 * 1000;
     while (leftover > 0) {
       const start = Date.now();
-      if (!isSourceInitiated(receipt))
-        throw new Error("Transfer not initiated");
 
-      const txstatus = await getTransactionStatus(
-        receipt.originTxs[receipt.originTxs.length - 1]!
-      );
+      if (
+        // this is awkward but there is not hasSourceInitiated like fn in sdk (todo)
```

Comment:
you just mean a func that does a gte check on `receipt.status`?

Reply:
right